pkg/apiserver/registry/virtualmachinepolicy/rest.go

```diff
@@ -76,24 +76,24 @@ func (r *REST) Get(ctx context.Context, name string, _ *metav1.GetOptions) (runt
 		return nil, errors.NewBadRequest("Namespace parameter required.")
 	}
 	fetchKey := types.NamespacedName{Namespace: ns, Name: name}
-	obj, found, _ := r.vmpIndexer.GetByKey(fetchKey.String())
-	if !found {
+	objs, err := r.npTrackerIndexer.ByIndex(networkpolicy.CloudResourceNpTrackerIndexerByNamespacedName, fetchKey.String())
+	if err != nil || len(objs) == 0 {
 		return nil, errors.NewNotFound(runtimev1alpha1.Resource("virtualmachinepolicy"), name)
 	}
-	return r.convertToVMP(obj.(*networkpolicy.NetworkPolicyStatus)), nil
+	return r.convertToVmp(objs[0]), nil
```

Comment:
Maybe also add a comment here explaining there can only be one matching obj?
Like same comment below `// For a given vm namespaced name there should be only one matching vm.`

Reply:
Moved the comment from convertToVmp() to here.